### PR TITLE
fixes #542 adding validation

### LINKED ISSUE #542: Implement validation object to FormConfig
Which validator should we be using (see #516)?  Validation should be an object.  object names should map directly to form elements.

Add Validation to form

On enter (onchange, onblur, etc)
Find out how the model works
https://angular.io/guide/forms
Control with config when the validation is done, on enter, on submit, etc.

### LINKED ISSUE #578: Add validation to bossy-form-textarea
Relates to #542 , basic validation for form-text-area, proof of concept

### PR DESCRIPTION
**Pull request by oliverrayreed**

### CODE REVIEW

File: src/bossy-ui/components/form/form.ts [hunk 1]
```diff
@@ -20,14 +20,32 @@ export class BossyFormComponent implements OnInit {
 
     this.isFormInlinedFromConfig = this.config.isFormInlined;
     this.config.elements.forEach((element) => {
-      const {name, value} = element;
-      elements[name] = value;
+      const {name, value, validators} = element;
+      // console.log(validators && validators[0] && validators[0].type);
```
Comment (tabersays) [inline src/bossy-ui/components/form/form.ts]: remove this comment

File: src/bossy-ui/components/form/form.ts [hunk 2]
```diff
@@ -20,14 +20,32 @@ export class BossyFormComponent implements OnInit {
...
+            } else if (Validators[validators[valids].type](validators[valids].value) != null) {
+              bossyValidators.push(Validators[validators[valids].type](validators[valids].value));
+            }
+          }
+        }
+      }
+      elements[name] = [value, bossyValidators];
     });
-
+    // console.log(elements)
```
Comment (tabersays) [inline src/bossy-ui/components/form/form.ts]: Remove this comment

File: src/bossy-ui/components/form/form.ts [hunk 3]
```diff
@@ -20,14 +20,32 @@ export class BossyFormComponent implements OnInit {
...
     this.bossyForm = this.formBuilder.group(elements);
+    console.log(this.bossyForm.controls['textInput'].valid);
```
Comment (tabersays) [inline src/bossy-ui/components/form/form.ts]: Remove this console

**Comment by glawson** [comment]
resolves #578

**Pull request merged**